File: 108-convert-sorted-array-to-binary-search-tree/convert-sorted-array-to-binary-search-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {number[]} nums
 * @return {TreeNode}
 */
var sortedArrayToBST = function(nums) {
    if(nums.length===0) return null;
    const buildTree=(left,right)=>{
        if (left >right) return null;
        let mid=Math.floor((left+right)/2), root=new TreeNode(nums[mid]);
        root.left=buildTree(left,mid-1);
        root.right=buildTree(mid+1,right);
        return root;
    };
    return buildTree(0,nums.length-1);
};

File: 108-convert-sorted-array-to-binary-search-tree/convert-sorted-array-to-binary-search-tree.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public TreeNode SortedArrayToBST(int[] nums) {
        return BuildTree(nums, 0, nums.Length - 1);
    }

    private TreeNode BuildTree(int[] nums, int left, int right) {
        if (left > right) {
            return null;
        }

        int mid = left + (right - left) / 2;

        TreeNode root = new TreeNode(nums[mid]);

        root.left = BuildTree(nums, left, mid - 1);

        root.right = BuildTree(nums, mid + 1, right);

        return root;
    }
}
